.circleci/config.yml

```diff
@@ -156,6 +160,35 @@ references:
         mkdir ${PACKAGES_DIR:?}
         mv _build/prod/rel/aeternity/aeternity-$(cat VERSION).tar.gz ${PACKAGE_TARBALL:?}
 
+  build_ubuntu_package: &build_ubuntu_package
+    run:
+      name: Build Ubuntu package (*.deb)
+      environment:
+        PACKAGES_DIR: *packages_workspace
+        # The *clean commands are for work-around in CircleCI
+        # Debian/Ubuntu package building (clean issues; fakeroot and
+        # rebar3).
+        #
+        # The debian/rules file has conditional *clean targets based on
+        # DEB_SKIP_DH_AUTO_CLEAN.
+        #
+        # This prevents the presence of broken/dummy debian/rules file without
+        # clean targets. For example when running debuild directly without
+        # `make prod-deb-package`.
+      command: |
+        make prod-clean
```

Comment:
It looks like this group of steps assumes clean initial state. If so, why placing commands for clean (that could stain) rather than ensuring in the place where this item is referenced that there is only checkout in a clean container?

Reply:
This is unfortunately an ugly hack, because of CircleCI. 

Debian/Ubuntu packaging tools and rules always run clean targets before building. However with our build container only in CircleCI, rebar3 has issues running clean form within the Debian wrappers (frakeroot especially). This is not observed on the same build container on a stand-alone machine with Docker in it.

This just mimics  as much as possible the Debian/Ubuntu packaging tasks when in CircleCI. Just to be safe. The Debian rules file does not run the clean targets when ran in CircleCI. 

I see no other work-around. It is at all not clear why container behaves like this in CircleCI. 

I think I am not confusing and did not forget where and when this happens, since the conclusion was made a while ago.

Reply:
What Luca meant that we don't need the extra clean steps here, as the workdir is clean already